src/util/CancellationHandle.h

```diff
@@ -81,12 +82,13 @@ struct PseudoStopToken {
 class CancellationException : public std::runtime_error {
  public:
   using std::runtime_error::runtime_error;
-  CancellationException(CancellationState reason, std::string_view details)
-      : std::runtime_error{absl::StrCat("Cancelled due to ",
-                                        reason == CancellationState::TIMEOUT
-                                            ? "timeout"
-                                            : "manual cancellation",
-                                        ". Stage: ", details)} {
+  CancellationException(CancellationState reason,
+                        ad_utility::source_location location)
+      : std::runtime_error{absl::StrCat(
+            "Cancelled due to ",
+            reason == CancellationState::TIMEOUT ? "timeout"
+                                                 : "manual cancellation",
+            " at ", location.file_name(), ":", location.line())} {
```

Comment:
I think we should trim the filename to make the log fit into one line (we currently get full paths there). Trim it to the last k bytes with a `...` before it or something.

Reply:
Okay, probably we don't want the source location at all here, as this is a user-facing exception which would leak internal paths into the UI which is ugly. Just say cancelled because of [timeout/manual cancellation].